app/models/order.rb

```diff
@@ -0,0 +1,72 @@
+class Order < ApplicationRecord
+  belongs_to :user
+  has_many :order_books
+  has_many :books, through: :order_books
+
+  extend Enumerize
+  enumerize :status, in: Constants::ORDER_STATUS, predicates: true, scope: true
+
+  validates :shipping_address, presence: true
+  validates :total_price, presence: true
+
+  before_create :generate_id
+  after_update_commit :send_emails_for_confirmed_order, if: :fresh_confirmed?
+  after_update_commit :send_emails_for_cancelled_order, if: :fresh_cancelled?
+
+  def fresh_confirmed?
+    self.previous_changes.has_key?(:status) && self.confirmed?
+  end
+
+  def fresh_cancelled?
+    self.previous_changes.has_key?(:status) && self.cancelled?
+  end
+
+  def display_created_at
+    created_at.strftime(Constants::FULL_TIME_FORMAT_FOR_DISPLAY).squish
+  end
+
+  def available_for_cancel?
+    if status.in?(%w[completed failed cancelled]) ||
+      (status == :confirmed && created_at < Time.zone.now - Constants::ORDER_CANCELLATION_BUFFER_HOURS.hours)
```

Comment:
Nit: can we use `Time.current` instead of `Time.zone.now`?

Reply:
Yes, we can. I check some articles and some best practices about it, The `Time.current`  is no different except cleaner/shorter syntax `Time.zone.now`, is it right?

Reply:
Yes, there should be no different.